### PR TITLE
Prevent field guide related requests if no field guide

### PR DESCRIPTION
**Pull request by mcbouslog**
Staging branch URL: https://fix-get-all-linked.pfe-preview.zooniverse.org/
- staged branch, project no Field Guide: https://fix-get-all-linked.pfe-preview.zooniverse.org/projects/markb-panoptes/calbug
- same project no Field Guide: https://master.pfe-preview.zooniverse.org/projects/markb-panoptes/calbug

Fixes annoying console error `Uncaught (in promise) Error: getAllLinked: resource and link not specified` in any project without a Field Guide.

Check if `guide` is defined before requesting all linked `attached_images`.

# Required Manual Testing

- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Comment by mcbouslog** [comment]
Very minor PR, but has been annoying me.

File: app/pages/project/index.jsx [hunk 1]
```diff
@@ -293,12 +293,14 @@ class ProjectPageController extends React.Component {
       this.setState({ guide });
       const guideId = guide ? guide.id : undefined;
       actions.translations.load('field_guide', guideId, translations.locale);
```
Comment (eatyourgreens) [inline app/pages/project/index.jsx]: Probably shouldn’t load translations for an undefined guide either.

**Comment by mcbouslog** [comment]
Admitting my ignorance in regards to translations here, but Is there a project on staging with translations (I suppose specifically of a Field Guide) I could test this on to make sure I didn't break anything? Or instructions on a quick way to test accordingly?

**Comment by eatyourgreens** [comment]
Snapshot Hoge Veluwe is available in English and Dutch.

**Pull request merged**